spec/lib/secure_headers/headers/policy_management_spec.rb

```diff
@@ -142,17 +147,33 @@ module SecureHeaders
           ContentSecurityPolicy.validate_config!(ContentSecurityPolicyConfig.new(default_opts.merge(plugin_types: ["application/pdf"])))
         end.to_not raise_error
       end
+
+      it "doesn't allow report_only to be set in a non-report-only config" do
+        expect do
+          ContentSecurityPolicy.validate_config!(ContentSecurityPolicyConfig.new(default_opts.merge(report_only: true)))
+        end.to raise_error(ContentSecurityPolicyConfigError)
+      end
+
+      it "allows report_only to be set in a report-only config" do
+        expect do
+          ContentSecurityPolicy.validate_config!(ContentSecurityPolicyReportOnlyConfig.new(default_opts.merge(report_only: true)))
+        end.to_not raise_error
+      end
     end
 
     describe "#combine_policies" do
+      before(:each) do
+        reset_config
```

Comment:
For groups of tests that want to play with the default configuration we need to reset it so we don't get an `AlreadyConfiguredError`